### PR TITLE
dogapi: fix response type of send_all

### PR DESCRIPTION
**Pull request by pabo**
## explanation
The dogapi `metric.send_all` has an incorrect type for the `res` in the callback. It should be an `EventCreateResponse`.

## template
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://brettlangdon.github.io/node-dogapi/#metric-send_all
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## dogapi ([<kbd>unpkg</kbd>](https://unpkg.com/browse/dogapi@latest/))
was missing the following properties:
1. client
1. now
1. OK
1. WARNING
1. CRITICAL

<details>
<summary>as well as these 1 other properties...</summary>
<p>UNKNOWN</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e70da96142e7e8461ab5a2e8159e355d2525ab64
</p>

File: types/dogapi/dogapi-tests.ts [hunk 1]
```diff
@@ -22,11 +22,13 @@ dogapi.event.create(
...
-        metric: 'metricName',
-        points: 500,
-    }],
-    (err: Error | null, res: string) => {}
+    [
+        {
+            metric: 'metricName',
+            points: 500,
+        },
+    ],
```
Comment (olebedev) [inline types/dogapi/dogapi-tests.ts:30]: I am not a big fan of merging style fixes & logic fixes all together. Can we do it in another PR if you really unhappy about it please?

**Review by olebedev — CHANGES_REQUESTED** [review]

File: types/dogapi/dogapi-tests.ts (hunk 1 above)
Comment (pabo) [reply types/dogapi/dogapi-tests.ts:30]: Overzealous auto-formatting extensions. I'll separate it out.

Comment (pabo) [reply types/dogapi/dogapi-tests.ts]: @olebedev please see latest commit.

I will submit further PR to align the formatting of these files with the main repo's formatting https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/.prettierrc.json

**Review by olebedev — APPROVED** [review]
LGTM 👍

**Comment by olebedev** [comment]
Ready to merge
